packages/java/src/node/java-contribution.ts

```diff
@@ -22,59 +22,56 @@ configurations.set('linux', 'config_linux');
-        const jarPaths = glob.sync('**/plugins/org.eclipse.equinox.launcher_*.jar', { cwd: serverPath });
-        if (jarPaths.length === 0) {
-            throw new Error('The Java server launcher is not found.');
-        }
+  start(clientConnection: IConnection): void {
+    const serverPath = path.resolve(__dirname, 'server');
+    const jarPaths = glob.sync('**/plugins/org.eclipse.equinox.launcher_*.jar', { cwd: serverPath });
+    if (jarPaths.length === 0) {
+      throw new Error('The Java server launcher is not found.');
+    }
 
-        const jarPath = path.resolve(serverPath, jarPaths[0]);
-        const workspacePath = path.resolve(os.tmpdir(), '_ws_' + new Date().getTime());
-        const configuration = configurations.get(process.platform);
-        if (!configuration) {
-            throw new Error('Cannot find Java server configuration for ' + process.platform);
-        }
-        const configurationPath = path.resolve(serverPath, configuration);
-        const command = 'java';
-        const args = [
-            '-Declipse.application=org.eclipse.jdt.ls.core.id1',
-            '-Dosgi.bundles.defaultStartLevel=4',
-            '-Declipse.product=org.eclipse.jdt.ls.core.product'
-        ];
+    const jarPath = path.resolve(serverPath, jarPaths[0]);
+    const workspacePath = path.resolve(os.tmpdir(), '_ws_' + new Date().getTime());
+    const configuration = configurations.get(process.platform);
+    if (!configuration) {
+      throw new Error('Cannot find Java server configuration for ' + process.platform);
+    }
+    const configurationPath = path.resolve(serverPath, configuration);
+    const command = 'java';
+    const args = [
+      '-Declipse.application=org.eclipse.jdt.ls.core.id1',
+      '-Dosgi.bundles.defaultStartLevel=4',
+      '-Declipse.product=org.eclipse.jdt.ls.core.product'
+    ];
 
-        if (DEBUG_MODE) {
-            args.push(
-                '-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1044',
-                '-Dlog.protocol=true',
-                '-Dlog.level=ALL'
-            );
-        }
+    if (DEBUG_MODE) {
+      args.push(
+        '-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=1044',
+        '-Dlog.protocol=true',
+        '-Dlog.level=ALL'
+      );
+    }
 
-        args.push(
-            '-jar', jarPath,
-            '-configuration', configurationPath,
-            '-data', workspacePath
-        );
+    args.push(
+      '-jar', jarPath,
+      '-configuration', configurationPath,
+      '-data', workspacePath
+    );
 
-        Promise.all([
-            this.startSocketServer(), this.startSocketServer()
-        ]).then(servers => {
-            const [inServer, outServer] = servers;
-            const inSocket = this.accept(inServer);
-            const outSocket = this.accept(outServer);
+    Promise.all(
+      [this.startSocketServer()]
+    ).then(servers => {
+      const [server] = servers;
```

Comment:
Minor: perhaps, you could get rid of the array of servers and the `Promise.all` as well if we have one, right? And write 
```typescript
this.startSocketServer().then(server => {
  // blabla
});
```
instead.